app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,84 @@
+class TweetsController < ApplicationController
+  
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    if  @user
+      erb :'tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = User.find(session[:id])
+      erb :'tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets' do
+    if !logged_in?
+      redirect '/login'
+    else
+      @user = User.find(session[:id])
+      erb :home
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :'tweets/edit_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  patch '/tweets/:id/edit' do
+    @old_tweet = Tweet.find(params[:id].to_i)
+    if params[:content] != @old_tweet && !params[:content].empty?
+      @old_tweet.content = params[:content]
+    end
+    @old_tweet.save
+  end
+
+  post '/tweets' do
+    @user = User.find(session[:id])
+    if !params[:content].empty?
+      @user.tweets << Tweet.create(content: params[:content])
+    else
+      redirect '/tweets/new'
+    end
+    @user.save
+    redirect "/tweets/#{@user.id}"
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+      session[:id] = @user.id
+      redirect to '/tweets'
+  end
+
+
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find(params[:id])
+    if session[:id] == @tweet.user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.